osu.Game/Configuration/SettingsStore.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Game.Database;
+
+namespace osu.Game.Configuration
+{
+    public class SettingsStore : DatabaseBackedStore
+    {
+        public event Action SettingChanged;
+
+        public SettingsStore(Func<OsuDbContext> createContext)
+            : base(createContext)
+        {
+        }
+
+        /// <summary>
+        /// Retrieve <see cref="DatabasedSetting"/>s for a specified ruleset/variant content.
+        /// </summary>
+        /// <param name="rulesetId">The ruleset's internal ID.</param>
+        /// <param name="variant">An optional variant.</param>
+        /// <returns></returns>
+        public List<DatabasedSetting> Query(int? rulesetId = null) =>

```

Comment:
We're going to want per-variant settings (e.g. different scrollspeeds on 4K and 7K).